File: app/src/main/java/com/user/canopas/dagger/mvp/presenter/ActorPresenter.java
package com.user.canopas.dagger.mvp.presenter;


import android.util.Log;

import com.user.canopas.dagger.mvp.view.ActorView;
import com.user.canopas.dagger.Api.ApiService;
import com.user.canopas.dagger.mvp.model.ActorRespone;


import javax.inject.Inject;

import rx.Observable;
import rx.Observer;

public class ActorPresenter extends BasePresenter<ActorView> implements Observer<ActorRespone> {

    @Inject
    ApiService mApiService;


    public void getActor() {
        Observable<ActorRespone> observable = mApiService.getActors();
        subscribe(observable, this);
    }

    @Inject
    public ActorPresenter() {
    }




    @Override
    public void onCompleted() {
        getView().compelet();
    }

    @Override
    public void onError(Throwable e) {

        getView().Error(e.getMessage());
    }

    @Override
    public void onNext(ActorRespone actorRespone) {
        getView().GetActorResponse(actorRespone);
    }

}

File: app/src/main/java/com/user/canopas/dagger/mvp/presenter/ActorPresenter.kt
package com.user.canopas.dagger.mvp.presenter


import android.util.Log

import com.user.canopas.dagger.mvp.view.ActorView
import com.user.canopas.dagger.Api.ApiService
import com.user.canopas.dagger.mvp.model.ActorRespone


import javax.inject.Inject

import rx.Observable
import rx.Observer

class ActorPresenter @Inject
constructor() : BasePresenter<ActorView>(), Observer<ActorRespone> {

    @Inject
    @JvmField var mApiService: ApiService? = null


    fun getActor() {
        val observable = mApiService!!.actors
        subscribe(observable, this)
    }


    override fun onCompleted() {
        view?.compelet()
    }

    override fun onError(e: Throwable) {

        view?.Error(e.message)
    }

    override fun onNext(actorRespone: ActorRespone) {
        view?.GetActorResponse(actorRespone)
    }

}
